src/MainWindow.vala

```diff
@@ -364,48 +371,40 @@ public class Camera.MainWindow : Hdy.ApplicationWindow {
     }
 
     private void add_camera_option (Gst.Device camera) {
-        var menuitem = new Gtk.RadioMenuItem.with_label (null, camera.display_name);
-        menuitem.set_data<Gst.Device> ("camera", camera);
-        camera_options.append (menuitem);
-
-        int i = (int) camera_options.get_children ().length () - 1;
-        if (i > 0) {
-            var el = camera_options.get_children ().nth_data (0) as Gtk.RadioMenuItem;
-            menuitem.join_group (el);
+        var menuitem = new MenuItem (camera.display_name, null);
+        menuitem.set_action_and_target (ACTION_PREFIX + ACTION_CHANGE_CAMERA, "s", camera.name);
+        camera_options.set_data<Gst.Device> (camera.name, camera);
+        camera_options.append_item (menuitem);
+
+        if (change_camera_action != null ) {
+            change_camera_action.set_state (new Variant.string (camera.name));
         }
-        menuitem.active = true;
-        menuitem.activate.connect (() => {
-            if (menuitem.active) {
-                camera_view.change_camera (menuitem.get_data<Gst.Device> ("camera"));
-            }
-        });
-        menuitem.show ();
 
         update_take_button ();
         enable_header (true);
     }
 
+    private void on_change_camera (GLib.SimpleAction action, GLib.Variant? parameter) {
+        action.set_state (parameter);
+        camera_view.change_camera (camera_options.get_data<Gst.Device> (parameter.get_string ()));
+    }
+
     private void remove_camera_option (Gst.Device camera) {
-        Gtk.Widget to_remove = null;
-        foreach (unowned Gtk.Widget menuitem in camera_options.get_children ()) {
-            var name = ((Gtk.MenuItem) menuitem).get_data<Gst.Device> ("camera").name;
-            if (name == camera.name) {
-                to_remove = menuitem;
-                break;
+        var item_count = camera_options.get_n_items ();
+        for (var index = 0; index < item_count; index++) {
+            var variant = camera_options.get_item_attribute_value (index, Menu.ATTRIBUTE_TARGET, new VariantType ("s"));
+            if (variant.get_string () == camera.name) {
+                camera_options.remove (index);
```

Comment:
```suggestion
            if (variant.get_string () == camera.name) {
                camera_options.remove (index);
                item_count--;
                break;
```
Lets reuse the `item_count` here to use in `enable_header()`. also break the for loop, we don't need to iterate it anymore.